rcl_action/test/rcl_action/test_action_interaction.cpp

```diff
@@ -0,0 +1,906 @@
+      &this->action_server,
+      &num_subscriptions_server,
+      &num_guard_conditions_server,
+      &num_timers_server,
+      &num_clients_server,
+      &num_services_server);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_action_client_wait_set_get_num_entities(
+      &this->action_client,
+      &num_subscriptions_client,
+      &num_guard_conditions_client,
+      &num_timers_client,
+      &num_clients_client,
+      &num_services_client);
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_wait_set_init(
+      &this->wait_set,
+      num_subscriptions_server + num_subscriptions_client,
+      num_guard_conditions_server + num_guard_conditions_client,
+      num_timers_server + num_timers_client,
+      num_clients_server + num_clients_client,
+      num_services_server + num_services_client,
+      rcl_get_default_allocator());
+    ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    // Finalize
+    rcl_ret_t ret = rcl_action_server_fini(&this->action_server, &this->node);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_clock_fini(&this->clock);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_action_client_fini(&this->action_client, &this->node);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_node_fini(&this->node);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_wait_set_fini(&this->wait_set);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+    ret = rcl_shutdown(&context);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  }
+
+  void init_test_uuid0(uint8_t * uuid)
+  {
+    for (uint8_t i = 0; i < UUID_SIZE; ++i) {
+      uuid[i] = i;
+    }
+  }
+
+  void init_test_uuid1(uint8_t * uuid)
+  {
+    for (uint8_t i = 0; i < UUID_SIZE; ++i) {
+      uuid[i] = 15 - i;
+    }
+  }
+
+  rcl_action_client_t action_client;
+  rcl_action_server_t action_server;
+  rcl_node_t node;
+  rcl_clock_t clock;
+  rcl_context_t context;
+
+  rcl_wait_set_t wait_set;
+
+  bool is_goal_request_ready;
+  bool is_cancel_request_ready;
+  bool is_result_request_ready;
+  bool is_goal_expired;
+
+  bool is_feedback_ready;
+  bool is_status_ready;
+  bool is_goal_response_ready;
+  bool is_cancel_response_ready;
+  bool is_result_response_ready;
+};  // class TestActionClientServerInteraction
+
+// Exercises the "Example 1" sequence diagram found in the actions_proposal document.
+// In this example, the action client request a goal and gets a response from
+// the server accepting the goal (synchronous). Upon accepting the goal, the
+// action server starts a user defined execution method for completing the goal.
+// Following the goal request, the client makes an asynchronous request for the
+// result. The user defined method publishes feedback to the action client as it
```

Comment:
The sentences about the "user defined method" are not really relevant to `rcl`, but are referring to client library implementations built on top of `rcl`. We can probably omit that part of the comment.

Reply:
you're right. done, thanks!